test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -202,54 +203,54 @@ void ProtoConverter::visit(BinaryOp const& _x)
 
 void ProtoConverter::visit(VarDecl const& _x)
 {
-	m_output << "let x_" << m_numLiveVars << " := ";
+	string varName = "x_" + to_string(counter());
+	m_output << "let " << varName << " := ";
 	visit(_x.expr());
-	m_numVarsPerScope.top()++;
-	m_numLiveVars++;
 	m_output << "\n";
+	m_scopes.top().insert(varName);
+	m_variables.push_back(varName);
 }
 
 void ProtoConverter::visit(EmptyVarDecl const&)
 {
-	m_output << "let x_" << m_numLiveVars++ << "\n";
-	m_numVarsPerScope.top()++;
+	string varName = "x_" + to_string(counter());
+	m_output << "let " << varName << "\n";
+	m_scopes.top().insert(varName);
+	m_variables.push_back(varName);
 }
 
 void ProtoConverter::visit(MultiVarDecl const& _x)
 {
 	size_t funcId = (static_cast<size_t>(_x.func_index()) % m_functionVecMultiReturnValue.size());
 
-	int numInParams = m_functionVecMultiReturnValue.at(funcId).first;
-	int numOutParams = m_functionVecMultiReturnValue.at(funcId).second;
+	unsigned numInParams = m_functionVecMultiReturnValue.at(funcId).first;
+	unsigned numOutParams = m_functionVecMultiReturnValue.at(funcId).second;
 
 	// Ensure that the chosen function returns at least 2 and at most 4 values
 	yulAssert(
 		((numOutParams >= 2) && (numOutParams <= 4)),
 		"Proto fuzzer: Multi variable declaration calls a function with either too few or too many output params."
 	);
 
-	// We must start variable numbering past the number of live variables at this point in time.
-	// This creates let x_p,..., x_k :=
-	// (k-p)+1 = numOutParams
-	m_output <<
-		"let " <<
-		dev::suffixedVariableNameList("x_", m_numLiveVars, m_numLiveVars + numOutParams) <<
-		" := ";
+	// Obtain variable name suffix
+	unsigned startIdx = counter();
+	m_output << "let ";
+	vector<string> varsVec = createVars(startIdx, startIdx + numOutParams);
+	m_output << " := ";
 
 	// Create RHS of multi var decl
 	m_output << "foo_" << functionTypeToString(NumFunctionReturns::Multiple) << "_" << funcId;
 	m_output << "(";
 	visitFunctionInputParams(_x, numInParams);
 	m_output << ")\n";
-	// Update live variables in scope and in total to account for the variables created by this
-	// multi variable declaration.
-	m_numVarsPerScope.top() += numOutParams;
-	m_numLiveVars += numOutParams;
+	// Add newly minted vars in the multidecl statement to current scope
+	addToScope(varsVec);
 }
 
 void ProtoConverter::visit(TypedVarDecl const& _x)
 {
-	m_output << "let x_" << m_numLiveVars;
+	string varName = "x_" + to_string(counter());
```

Comment:
Minor suggestion that *could* be applied in several places throughout this PR.
```suggestion
	string varName{"x_" + to_string(counter())};
```

Reply:
Thank you :-)

Note to self: Prefer list initialization where possible because it errors on implicit conversions https://stackoverflow.com/a/18222927